File: Set 8/3 Sums/ThreeSum.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

bool three_sum(vector<int>& arr, int s) {
    int n = arr.size();
    sort(arr.begin(), arr.end());
    for (int i = 0; i < n - 2; i++) {
        int left = i + 1;
        int right = n - 1;
        while (left < right) {
            int sum = arr[i] + arr[left] + arr[right];
            if (sum == s) {
                cout << arr[i] << " " << arr[left] << " " << arr[right] << endl;
                return true;
            } else if (sum < s) {
                left++;
            } else {
                right--;
            }
        }
    }
    return false;
}

int main() {
    int t;
    cout << "Enter Test Case Count: ";
    cin >> t;

    while (t--) {
        int n, s;
        cout << "Enter the size of the array: ";
        cin >> n;
        vector<int> arr(n);
        cout << "Enter the array elements: ";
        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }
        cout << "Enter the target sum: ";
        cin >> s;

        if (!three_sum(arr, s)) {
            cout << "false" << endl;
        }
    }
    return 0;
}

File: Set 8/3 Sums/ThreeSum.py
def three_sum(arr, s):
    arr.sort()
    n = len(arr)
    for i in range(n - 2):
        left, right = i + 1, n - 1
        while left < right:
            total = arr[i] + arr[left] + arr[right]
            if total == s:
                print(arr[i], arr[left], arr[right])
                return True
            elif total < s:
                left += 1
            else:
                right -= 1
    return False


t = int(input("Enter Test Case Count: "))
for _ in range(t):
    arr = list(map(int, input("Enter array elements: ").split()))
    s = int(input("Enter the target sum: "))
    if not three_sum(arr, s):
        print("false")


